migration-engine/connectors/sql-migration-connector/src/flavour/mssql.rs

```diff
@@ -216,119 +216,126 @@ impl SqlFlavour for MssqlFlavour {
-                EXEC SP_EXECUTESQL @stmt
-                "#,
-                schema_name
-            );
-
-            let drop_shared_defaults = format!(
-                r#"
-                DECLARE @stmt NVARCHAR(max)
-                DECLARE @n CHAR(1)
-
-                SET @n = CHAR(10)
-
-                SELECT @stmt = ISNULL(@stmt + @n, '') +
-                    'DROP DEFAULT [' + SCHEMA_NAME(schema_id) + '].[' + OBJECT_NAME(object_id) + ']'
-                FROM sys.objects
-                WHERE SCHEMA_NAME(schema_id) = '{0}' AND type = 'D' AND parent_object_id = 0
-
-                EXEC SP_EXECUTESQL @stmt
-                "#,
-                schema_name
-            );
-
-            let drop_views = format!(
-                r#"
-                DECLARE @stmt NVARCHAR(max)
-                DECLARE @n CHAR(1)
-
-                SET @n = CHAR(10)
-
-                SELECT @stmt = ISNULL(@stmt + @n, '') +
-                    'DROP VIEW [' + SCHEMA_NAME(schema_id) + '].[' + name + ']'
-                FROM sys.views
-                WHERE SCHEMA_NAME(schema_id) = '{0}'
-
-                EXEC SP_EXECUTESQL @stmt
-                "#,
-                schema_name
-            );
-
-            let drop_fks = format!(
-                r#"
-                DECLARE @stmt NVARCHAR(max)
-                DECLARE @n CHAR(1)
-
-                SET @n = CHAR(10)
-
-                SELECT @stmt = ISNULL(@stmt + @n, '') +
-                    'ALTER TABLE [' + SCHEMA_NAME(schema_id) + '].[' + OBJECT_NAME(parent_object_id) + '] DROP CONSTRAINT [' + name + ']'
-                FROM sys.foreign_keys
-                WHERE SCHEMA_NAME(schema_id) = '{0}'
-
-                EXEC SP_EXECUTESQL @stmt
-                "#,
-                schema_name
-            );
-
-            let drop_tables = format!(
-                r#"
-                DECLARE @stmt NVARCHAR(max)
-                DECLARE @n CHAR(1)
-
-                SET @n = CHAR(10)
-
-                SELECT @stmt = ISNULL(@stmt + @n, '') +
-                    'DROP TABLE [' + SCHEMA_NAME(schema_id) + '].[' + name + ']'
-                FROM sys.tables
-                WHERE SCHEMA_NAME(schema_id) = '{0}'
-
-                EXEC SP_EXECUTESQL @stmt
-                "#,
-                schema_name
-            );
-
-            let drop_types = format!(
-                r#"
-                DECLARE @stmt NVARCHAR(max)
-                DECLARE @n CHAR(1)
-
-                SET @n = CHAR(10)
-
-                SELECT @stmt = ISNULL(@stmt + @n, '') +
-                    'DROP TYPE [' + SCHEMA_NAME(schema_id) + '].[' + name + ']'
-                FROM sys.types
-                WHERE SCHEMA_NAME(schema_id) = '{0}'
-                AND is_user_defined = 1
-
-                EXEC SP_EXECUTESQL @stmt
-                "#,
-                schema_name
-            );
+            let ns_vec = Namespaces::to_vec(namespaces, params.url.schema().to_string());
+            for schema_name in &ns_vec {
```

Comment:
This will be broken in some cases:

Let's imagine you have tables `Blog` and `User`.

- `User` has a foreign key to `Blog`
- `User` is in schema `private` and `Blog` in schema `public`
- We iterate schemas, and we get `private` before `public`
- Trying to `DROP TABLE [Blog]`, we hit a runtime error, because the foreign key is in the `User` table and `public` schema.

What we should do here is:

- iterate all namespaces, delete only procedures
- iterate all namespaces again, delete only the defaults
- ... and so on

This way we can be sure we have all items depending on some other items dropped before we hit any runtime errors.